demisto_sdk/commands/generate_modeling_rules/generate_modeling_rules.py

```diff
@@ -0,0 +1,439 @@
+        xdm_field_type,
+        xdm_class_type,
+        is_array_raw,
+        type_raw,
+    ):
+        """
+        xdm_rule - The xdm rule
+        field_path_raw - the path to the field in the raw event
+        field_type - the field type in the schema.
+        xdm_class_type - xdm rule type
+        is_array_raw - is the raw event of type array.
+        type_raw - the type of the raw event field.
+        """
+        self.xdm_rule = xdm_rule
+        self.field_path_raw = field_path_raw
+        self.xdm_field_type = xdm_field_type
+        self.xdm_class_type = xdm_class_type
+        self.is_array_raw = is_array_raw
+        self.type_raw = type_raw
+
+    def create_schema_types(self) -> dict:
+        return {"type": self.type_raw, "is_array": self.is_array_raw}
+
+
+def to_string(s: str) -> str:
+    """
+    Gets a xql and wraps it with a to_string function
+    """
+    return f"to_string({s})"
+
+
+def to_number(s: str) -> str:
+    """
+    Gets a xql and wraps it with a to_number function
+    """
+    return f"to_number({s})"
+
+
+def json_extract_array(prefix: str, suffix: str) -> str:
+    return f'json_extract_array({prefix}, "$.{suffix}")'
+
+
+def json_extract_scalar(prefix: str, suffix: str) -> str:
+    return f'json_extract_scalar({prefix}, "$.{suffix}")'
+
+
+def array_create(s: str) -> str:
+    return f"arraycreate({s})"
+
+
+def create_xif_header(dataset_name: str) -> str:
+    """
+    Creates the xif header
+    """
+    xif_rule = ""
+    xif_rule += f"[MODEL: dataset={dataset_name}]\n"
+    xif_rule += "| alter\n"
+    return xif_rule
+
+
+def init_mapping_field_list(
+    modeling_rules_df: pd.DataFrame,
+    raw_event: dict,
+    xdm_rule_to_dtype: dict,
+    xdm_rule_to_dclass: dict,
+) -> List[MappingField]:
+    """
+    This function takes all the data gathered and generates the list of MappingFields
+    """
+    name_columen = modeling_rules_df["Name"]
+    xdm_one_data_model = modeling_rules_df["XDM Field One Data Model"]
+    names_list = name_columen.to_numpy()
+    xdm_one_data_model_list = xdm_one_data_model.to_numpy()
+
+    mapping_list = []
+    for (field_name, xdm_field_name) in zip(names_list, xdm_one_data_model_list):
+        type_raw, is_array_raw = extract_raw_type_data(raw_event, field_name)
+        xdm_field_type = xdm_rule_to_dtype.get(xdm_field_name)
+        xdm_class_type = xdm_rule_to_dclass.get(xdm_field_name)
+
+        mapping_list.append(
+            MappingField(
+                xdm_rule=xdm_field_name,
+                field_path_raw=field_name,
+                xdm_field_type=xdm_field_type,
+                xdm_class_type=xdm_class_type,
+                is_array_raw=is_array_raw,
+                type_raw=type_raw,
+            )
+        )
+
+    return mapping_list
+
+
+def convert_raw_type_to_xdm_type(schema_type: str) -> str:
+    """
+    returns the xdm type convention
+    """
+    converting_dict = {"string": "String", "int": "Number", "boolean": "Boolean"}
```

Comment:
please use constants for `String`, `Number` etc, anywhere in this file.

Reply:
Done